tests/cupy_tests/random_tests/test_generator.py

```diff
@@ -253,6 +253,96 @@ def test_goodness_of_fit_2(self):
         self.assertTrue(hypothesis.chi_square_test(counts, expected))
 
 
+@testing.parameterize(
+    {'a': 3, 'size': 2, 'p': None},
+    {'a': 3, 'size': 2, 'p': [0.3, 0.3, 0.4]},
+    {'a': 3, 'size': (5, 5), 'p': [0.3, 0.3, 0.4]},
+    {'a': 3, 'size': (), 'p': None},
+    {'a': [1, 2, 3], 'size': 2, 'p': [0.3, 0.3, 0.4]},
```

Comment:
Do you have special reason why you chose `[1, 2, 3]` as the value of `a`? If nothing, how about changing the value of `a` in the last test case from `[1, 2, 3]` to `[0, 1, 2]`? It will remove cumbersome branchings in e.g. [here](https://github.com/pfnet/chainer/pull/1979/files?diff=unified#diff-d5961550728ebd08cfb44c0e5074cd97R281).